experimental/class-features.md

```diff
@@ -0,0 +1,63 @@
+
+# Class Features
+
+This experimental extension covers three class features proposals: 
+[Class Fields], [Static Class Features] and [Private Methods].
+
+## ClassBody
+
+```js
+extend interface ClassBody <: Node {
+    body: [ MethodDefinition | PropertyDefinition ];
+}
+```
+
+## PropertyDefinition
+
+```js
+interface PropertyDefinition <: Node {
+    type: "PropertyDefinition";
+    key: Expression | PrivateName;
+    value: Expression;
+    computed: boolean;
+    static: boolean;
+    private: boolean;
+}
+```
+
+- When `private` is `true`, `computed` must be `false` and `key` must be a `PrivateName`.
+- When `private` is `false`, `key` must be an `Expression` and can not be a `PrivateName`.
+
+## MethodDefinition
+
+```js
+extend interface MethodDefinition <: Node {
+    key: Expression | PrivateName;
+    static: boolean;
+    private: boolean;
+}
+```
+
+- When `private` is `true`, `computed` must be `false`, `key` must be a `PrivateName`, `kind` can not be `constructor`.
+- When `private` is `false`, `key` must be an `Expression` and can not be a `PrivateName`.
+
+### PrivateName
+
+```js
+interface PrivateName <: Node {
+    type: "PrivateName";
+    id: Identifier;
```

Comment:
How about just `name: string`? Because the [PrivateIdentifier](https://tc39.es/proposal-class-fields/#prod-PrivateIdentifier) production is a token that we cannot split it anymore, I feel odd using the `Identifier` node here.

Reply:
Related: https://github.com/babel/babylon/pull/260#pullrequestreview-38805907